src/compiler/checker.ts

```diff
@@ -11825,8 +11825,15 @@ namespace ts {
                                 }
                             }
                         }
-                        checkFlags |= (isReadonlySymbol(prop) ? CheckFlags.Readonly : 0) |
-                            (!(modifiers & ModifierFlags.NonPublicAccessibilityModifier) ? CheckFlags.ContainsPublic : 0) |
+                        // For unions, prop is readonly if any definition of it is readonly

```

Comment:
these comments seem superfluous to me -- the intersection one might need a comment if it were on its own, but here it's clearly the inverse of the union case, and the union case is clear as-is